src/library_webgl.js

```diff
@@ -1463,7 +1463,7 @@ var LibraryGL = {
       if (GLctx.currentPixelUnpackBufferBinding) {
         GLctx['compressedTexImage2D'](target, level, internalFormat, width, height, border, imageSize, data);
       } else {
-        GLctx['compressedTexImage2D'](target, level, internalFormat, width, height, border, HEAPU8, data, imageSize);
+        if (imageSize) GLctx['compressedTexImage2D'](target, level, internalFormat, width, height, border, HEAPU8, data, imageSize);
```

Comment:
This change would break the following code:

```html
<html><body><script>
var canvas = document.createElement('canvas');
var gl = canvas.getContext('webgl2');
var t = gl.createTexture();
var s3tc = gl.getExtension('WEBGL_compressed_texture_s3tc');
gl.bindTexture(gl.TEXTURE_2D, t);
var HEAPU8 = new Uint8Array(16*1024*1024);
while(gl.getError()) /*nop*/;
gl.compressedTexImage2D(gl.TEXTURE_2D, 0, s3tc.COMPRESSED_RGBA_S3TC_DXT1_EXT, 0, 0, 0, HEAPU8, 0, 0);
console.error(gl.getError());
</script></body></html>
```
This code is expected to get an error. E.g. it could be a genuine bug that caller of the code did. Why should we silently sweep the error under the rug?

Reply:
This seems like a vanishingly unlikely use - applications should not expect errors. C++ developers expect zero-upload calls to effectively be no-ops; however, the garbage-free entry points added in WebGL 2.0 use "0" for the length to mean "consume the entire buffer". We could possibly have decided something different during the spec's development, like having "undefined" mean something different than "0" - but at this point it's too late to make a breaking change.

I think that no-op'ing these calls in Emscripten is a good alternative.

Reply:
I agree it's not worthwhile to add quite a bit of code just to get exactly the proper errors. We could do it for FULL_ES2 but I don't think it's even worth the maintenance burden.